### DISCUSSION TITLE
Refit maintenance status

### DISCUSSION THREAD

**Discussion by hansmbakker** [reactiveui/refit#1447] [General]
Refit would be the first library that comes to mind for me when creating a .NET client for a REST API - it has a long track record and great functionality.
However, it was pointed out to me that activity on the Refit repo has dropped - on Reddit people were [mentioning Refit's repo inactivity](https://www.reddit.com/r/dotnet/comments/xxgwco/any_fullfeatured_oauth_20_library/) after which it occurred to me that the last pull request merged to main was 9 months ago.

There have been enough suggestions for contributions (PRs and issues) since that time, but I reckon nobody with time has the permissions to review and merge those? I understand that life can happen, or that a shift in interests can happen, but I'm simply wondering whether there are any there future plans for Refit or is there a recommended fork that I should track?

Could anybody from ReactiveUI comment on the status of Refit? (tagging the top contributors - @anaisbetts @clairernovotny @Sergio0694 @james-s-tayler as the ReactiveUI org cannot be tagged)

**Comment by bennor** [comment]
Probably worth me being removed TBH. I haven't made a contribution in years and haven't written a line of C# in almost as long.

I love this little library but do not have the time to help out. Happy to make way for someone with the time/inclination.

🥲

**Reply by hansmbakker** [reply]
Done, sorry for disturbing :) I simply looked at https://github.com/reactiveui/refit/graphs/contributors and didn't notice when the commits were made..

**Reply by bennor** [reply]
It's fine! I've been meaning to say something for a while but it kept slipping my mind. Thanks for the reminder. 🙂

**Reply by hansmbakker** [reply]
And thank you for your contributions!

**Comment by james-s-tayler** [comment]
@hansmbakker I've been wondering the same thing for a while now. It's technically part of the .NET Foundation, so in theory it should be able to be maintained in perpetuity, which is in part the reason I started contributing to it, though activity this year has dropped in comparison to recent years.

I think ReactiveUI has a slack channel somewhere you could locate and try asking there as a way to get more visibility if no responses here from those who have permission to merge etc,?

**Reply by hansmbakker** [reply]
@james-s-tayler I posted the question in https://app.slack.com/client/T02AG6QJM/C02AJB872 - the answer was

> The status of Refit, is that it Does What I Want It To Do, and I don't really want it to Do Anything Else
Also, I'm not super keen on doing unpaid work in my early semi-retirement, so that other people's companies can make money


It's a shame because Refit is a great library, but I respect the points above.

**Comment by anaisbetts** [comment]
@hansmbakker What in particular do you want Refit to do that it currently does not?

**Reply by hansmbakker** [reply]
I do not have something that is missing or broken per say, but other people and me do notice the drop in activity on the repo which was very active in the past. For now I would still use it but I was triggered to pose this question because of the linked Reddit thread where people were questioning Refit's activity.

Besides that - I cannot see Refit's dependabot status with security PRs, but I think in general it is good if dependencies are kept up to date. I do notice that other dependabot PRs are left unmerged which does raise questions about its status (security and technical debt - now and in the future).

**Reply by hansmbakker** [reply]
(I am not blaming you or other maintainers here at all - I am thankful for your work - I am simply wondering what we can expect going forward. If the maintainers have other priorities, that is also an answer)

**Reply by hansmbakker** [reply]
@anaisbetts @clairernovotny besides a statement on whether dependabot's security PRs will be merged, I think it would also be fair to people submitting PRs.

The last PR was merged in February and newer contributions are left in limbo - if the repo is not maintained anymore (which can, again, be a valid choice) then I would say it is fair to everybody to make that choice explicit. Now people put effort in something that will be most likely be left dangling.

**Reply by squadwuschel** [reply]
Would be nice if, the requested contributions could be merged into the Repo or you could mention the inactivity of the Repo directly on the Readme, thats what me happed, I've implemented large code base with refit and did not see that its not longer supported.

**Reply by james-s-tayler** [reply]
What I don't get is that it's part of the .NET Foundation, so there should be all the governance required to pass it to new maintainers if needed, so that it can be maintained going forward rather than just kind of abandoned.

**Reply by maroonedsia** [reply]
As of now, Refit is not a part of .NET foundation anymore: https://dotnetfoundation.org/projects/current-projects

**Reply by glennawatson** [reply]
Refit is part of the ReactiveUI project, which is a DNF project.

**Reply by Sergio0694** [reply]
Just out of curiosity, is this explicitly mentioned in the DNF terms? Because last I checked, the DNF status was per-respository, not per-org and then transitively applied to all projects. It would seem a bit weird if that was the case.

**Reply by glennawatson** [reply]
@Sergio0694 I'll have to come back to you on that one. There are changes happening to the RxUI projects in general in regards to the DNF that we can't discuss yet.
